In [2]:
import tensorflow as tf
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

In [3]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(),
  tf.keras.layers.Dense(512, activation=tf.nn.relu),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation=tf.nn.softmax)
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

In [4]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test, y_test)

Epoch 1/5
60000/60000 [==============================] - 15s 250us/step - loss: 0.1993 - acc: 0.9415
Epoch 2/5
60000/60000 [==============================] - 17s 288us/step - loss: 0.0797 - acc: 0.9750
Epoch 3/5
60000/60000 [==============================] - 19s 308us/step - loss: 0.0524 - acc: 0.9834
Epoch 4/5
60000/60000 [==============================] - 17s 281us/step - loss: 0.0365 - acc: 0.9880
Epoch 5/5
10000/10000 [==============================] - 1s 63us/step


[0.07436988834261428, 0.9779]

this gives us a trained net with an accuracy of ~98% in just a couple of lines of code!